tensorflow/core/framework/allocator.h

```diff
@@ -152,6 +152,7 @@ class Allocator {
   // allocated by this allocator.
   virtual size_t RequestedSize(void* ptr) {
     CHECK(false) << "allocator doesn't track sizes";
+    return size_t(0);
```

Comment:
Is this necessary? `LogMessageFatal::~LogMessageFatal()` has `TF_ATTRIBUTE_NORETURN`, so the lack of a return statement shouldn't (AIUI) give an error. Or is this compiler-version-specific?

(Same goes for the other similar changes in here.)